kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/impl/HenkiloServiceImpl.java

```diff
@@ -181,6 +179,36 @@ private boolean isVahvastiTunnistettu(Henkilo henkilo) {
                 || henkilo.getHenkiloVarmentajas().stream().anyMatch(HenkiloVarmentaja::isTila);
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public LogInRedirectType logInRedirectByOidhenkilo(String oidHenkilo) {
+        Henkilo henkilo = this.henkiloDataRepository.findByOidHenkilo(oidHenkilo)
+                .orElseThrow(() -> new NotFoundException("Henkilo not found with oid " + oidHenkilo));
+        return this.getLoginRedirectType(henkilo);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public LogInRedirectType logInRedirectByUsername(String username) {
+        Henkilo henkilo = this.henkiloDataRepository.findByKayttajatiedotUsername(username)
+                .orElseThrow(() -> new NotFoundException("Henkilo not found with username " + username));
+        return this.getLoginRedirectType(henkilo);
+    }
+
+    protected LogInRedirectType getLoginRedirectType(Henkilo henkilo) {
+        boolean isVahvastiTunnistettu = this.isVahvastiTunnistettu(henkilo);
+        if(Boolean.FALSE.equals(isVahvastiTunnistettu)) {
+            return LogInRedirectType.STRONG_IDENTIFICATION;
+        }
+
+        LocalDateTime sixMonthsAgo = LocalDateTime.now().minusMonths(6);
+        if(henkilo.getSahkopostivarmennusAikaleima() == null || henkilo.getSahkopostivarmennusAikaleima().isBefore(sixMonthsAgo)) {
+            return LogInRedirectType.EMAIL_VERIFICATION;
+        }
+
+        return null;
```

Comment:
Optional on yleensä kivempi palautus kuin null. Rajapinnan olisi syytä palauttaa jokin 40x vastaus jos tulee tällainen tilanne.

Reply:
Paras vaihtoehto nyt, kun tätä uudelleen miettii, olisi varmaankin tehdä LogInRedirectType.NO_REDIRECT. Tätä null nyt tässä tarkoittaa ja se on ihan validi tilanne, eikä tällöin 40x virheen heittäminen olisi mielestäni paras tapa.